templates/tower-project.yaml

```diff
@@ -0,0 +1,338 @@
+                - "s3:DeleteObject"
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}"
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+
+  TowerForgeServiceUserAccessKey:
+    Type: AWS::IAM::AccessKey
+    Properties:
+      UserName: !Ref TowerForgeServiceUser
+
+  TowerProjectConfigurationSecret:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Name: !Sub "${AWS::StackName}-TowerProjectConfiguration"
+      Description: !Sub "Configuration for ${AWS::StackName} Project in Tower"
+      SecretString: !Sub
+        - >-
+          {
+            "forge_access_key_id":      "${ForgeAccessKeyId}",
+            "forge_secret_access_key":  "${ForgeSecretAccessKey}",
+            "bucket_name":              "${BucketName}",
+            "read_write_arns":          "${ReadWriteArns}",
+            "read_only_arns":           "${ReadOnlyArns}",
+            "vpc_id":                   "${VpcId}",
+            "subnet_ids":               "${SubnetIds}"
+          }
+        - ForgeAccessKeyId: !Ref TowerForgeServiceUserAccessKey
+          ForgeSecretAccessKey: !GetAtt TowerForgeServiceUserAccessKey.SecretAccessKey
+          BucketName: !Ref TowerBucket
+          ReadWriteArns: !Join [ ",", !Ref ReadWriteAccessArns ]
+          ReadOnlyArns: !Join [ ",", !Ref ReadOnlyAccessArns ]
+          VpcId: !ImportValue 
+            'Fn::Sub': '${AWS::Region}-nextflow-vpc-VPCId'
+          SubnetIds: !Join
+            - ","
+            - - !ImportValue 
+                'Fn::Sub': '${AWS::Region}-nextflow-vpc-PrivateSubnet'
+              - !ImportValue 
+                'Fn::Sub': '${AWS::Region}-nextflow-vpc-PrivateSubnet1'
+              - !ImportValue 
+                'Fn::Sub': '${AWS::Region}-nextflow-vpc-PrivateSubnet2'
+              - !ImportValue 
+                'Fn::Sub': '${AWS::Region}-nextflow-vpc-PrivateSubnet3'
+      KmsKeyId: !GetAtt EncryptionKeyStack.Outputs.Key
+  
+  TowerProjectConfigurationSecretPolicy:
+    Type: AWS::SecretsManager::ResourcePolicy
+    Condition: HasAnyAccessArns
+    Properties:
+      BlockPublicPolicy: True
+      SecretId: !Ref TowerProjectConfigurationSecret
+      ResourcePolicy:
+        Version: "2012-10-17"
+        Statement:
+        - Sid: GrantReadAccessToReadWriteAccessArns
+          Effect: Allow
+          Action: "secretsmanager:GetSecretValue"
+          Principal:
+            AWS: !Split
+              - ","
+              - !Join
+                - ","
+                - - !If
+                    - HasReadWriteAccessArns
+                    - !Join
+                      - ","
+                      - !Ref ReadWriteAccessArns
+                    - !Ref AWS::NoValue
+                  - !If
+                    - HasReadOnlyAccessArns
+                    - !Join
+                      - ","
+                      - !Ref ReadOnlyAccessArns
+                    - !Ref AWS::NoValue
+          Resource: "*"
```

Comment:
wouldn't this give one user access to another's secrets?  might be better to add a prefix to your secrets so you can restrict access to the prefix? something like this.. `Resource: "/${AWS::StackName}/*`

Reply:
I wouldn't expect a resource-based policy bound to a specific secret to have any effect on other secrets. That said, I replaced the `Resource` to be `!Ref TowerProjectConfigurationSecret`.